tests/tests_integration/test_api/test_datapoints.py

```diff
@@ -992,6 +1005,46 @@ def test_query_with_duplicates(self, retrieve_endpoints, one_mill_dps_ts, ms_bur
             assert len(res_lst.get(external_id=ts_string.external_id)) == 2
 
 
+@pytest.mark.parametrize(
+    "raw_granularity, granularity, aggregation, timezone",
+    (
+        ("hourly", "1d", "average", "Europe/Oslo"),
+        ("hourly", "1d", "sum", "Europe/Oslo"),
+        # ("hourly", "1month", "average", "Europe/Oslo"),
+        # ("hourly", "1month", "sum", "Europe/Oslo"),
+        # ("hourly", "1year", "average", "Europe/Oslo"),
+        # ("hourly", "1year", "sum", "Europe/Oslo"),
+    ),
+)
+def test_aggregate_timezone(
+    raw_granularity: str,
+    granularity: str,
+    aggregation: str,
+    timezone: str,
+    cognite_client: CogniteClient,
+    hourly_normal_dist: TimeSeries,
+    minutely_normal_dist: TimeSeries,
+):
+    # Arrange
+    time_series = {"hourly": hourly_normal_dist, "minutely": minutely_normal_dist}[raw_granularity]
+    raw_df = cognite_client.time_series.data.retrieve_dataframe(external_id=time_series.external_id)
+    pandas_agg = {"average": "mean"}.get(aggregation, aggregation)
+    pandas_gran = {"1d": "1D"}.get(granularity, granularity)
+    # Note pandas handles timezones correctly, which is why we use it for reference.
+    pandas_aggregate = raw_df.tz_localize("UTC").tz_convert(timezone).resample(pandas_gran).agg(pandas_agg)
```

Comment:
Yes, `pandas` handles timezones correctly for the raw data, but the aggregation function does not for `average`/`mean` (`sum` ofc is correct haha).

`pandas` mean:
$$\frac{1}{N} \sum_{i=1}^N x_i$$

`CDF` average: the integral over time:
$$\frac{1}{b-a} \int_{t=a}^{t=b}f(x)dx$$
...where `f(x)` is the function defined by either linear or step interpolating between the datapoints `x_i`, also considering the data point right before `t=a` and after `t=b`, unless we have data precisely at the boundary (`a, b`).

Reply:
I implemented a function for doing the aggregation correctly. Unfortunately, I had to make some assumptions, uniform index with known raw frequency. Not looked at step yet either. But it is a start.